### PR TITLE
COSS.io

### PR DESCRIPTION
**Pull request by frosty00**
still working on getting sign working correctly

### CODE REVIEW

File: js/coss.js [hunk 1]
```diff
@@ -0,0 +1,907 @@
...
+            const ticker = this.parseTicker (tickers[i]);
+            const symbol = ticker['symbol'];
+            result[symbol] = ticker;
+        }
+        return result;
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        let tickers = await this.fetchTickers ([ symbol ], params);
+        return tickers[symbol];
```
Comment (frosty00) [inline js/coss.js]: shouldn't this be tickers[marketId] ?

**Comment by graffias86** [comment]
Hello. ccxt guys. thank you for offering coss api on ccxt. on 406 line of coss.py, I think there is some small thing that should be fixed.
 baseId, quoteId = marketId.split('_')       ->   baseId, quoteId = marketId.split('-')

from '-' to '_'

**Comment by graffias86** [comment]
image
Can you help me fixing this error?

**Comment by kroitor** [comment]
> Can you help me fixing this error?

Yep, one sec, will upload one more fix for it shortly.

**Comment by graffias86** [comment]
> > Can you help me fixing this error?
> 
> Yep, one sec, will upload one more fix for it shortly.

Thank you. How can I know the timing when you upload it? Can you reply to this message?

**Comment by kroitor** [comment]
> Thank you. How can I know the timing when you upload it? Can you reply to this message?

Well, I already did as you see from this commit above https://github.com/ccxt/ccxt/commit/9f33d949bd1c2f37d90193b2a29d3236b0bc2e6f

You should actually receive a notification from GitHub if you're subscribed to this thread (top right section of this page). Anyway, I've already uploaded it and it will take 5-10 minutes for a new build to arrive, please update to version 1.18.58 in 10 minutes and it should work from there. Thx for reporting, let us know if you find anything else, will be happy to help!

**Comment by saurabBajgain** [comment]
Hello. ccxt guys. thank you for offering coss api on ccxt. line 465 coss.js at coss.fetchTickers()
TypeError: this.exchangeGetGetmarketsummaries is not a function.
what might be the reason behind this error ? and also i couldnot find exchangeGetGetmarketsummaries () defined in exchange.js .

**Comment by frosty00** [comment]
Hi @saurabBajgain, `coss.fetchTickers` is working for me. What version of ccxt are you using?

**Comment by saurabBajgain** [comment]
Hi @frosty00  Thankyou for your reply.
it was a bit old and i updated to latest  one (v 1.18.995) and coss.fetchTickers is ok  but now another problem while placing limit sell order;-
**Request** URL:https://trade.coss.io/c/api/v1/order/add
**Status Code**:403 
**Request Playload:**
{timestamp: 1564652880856, order_symbol: "TRX_ETH", order_size: "123.02", order_side: "SELL",…}
order_price:"0.000106"
order_side:"SELL"
order_size:"123.02"
order_symbol:"TRX_ETH"
recvWindow:10000000
timestamp:1564652880856
type:"limit"

Invalid CORS request 

### exception in console:->
**ExchangeNotAvailable: POST https://trade.coss.io/c/api/v1/order/add 403  Invalid CORS request (possible reasons: invalid API keys, bad or old nonce, exchange is down or offline, on maintenance, DDoS protection, rate-limiting)**

what might be the reason?

**Comment by kroitor** [comment]
@saurabBajgain do you have sufficient funds to place that order? If you have enough funds, then it might be related to this: https://github.com/ccxt/ccxt/issues/5591

**Pull request merged**